src/ID.cc

```diff
@@ -219,15 +219,35 @@ void ID::UpdateValAttrs()
 		}
 	}
 
-void ID::MakeDeprecated()
+void ID::MakeDeprecated(Expr* deprecation)
 	{
 	if ( IsDeprecated() )
 		return;
 
-	attr_list* attr = new attr_list{new Attr(ATTR_DEPRECATED)};
+	attr_list* attr = new attr_list{new Attr(ATTR_DEPRECATED, deprecation)};
 	AddAttrs(new Attributes(attr, Type(), false));
 	}
 
+string ID::GetDeprecationWarning() const
+	{
+	string result;
+	Attr* depr_attr = FindAttr(ATTR_DEPRECATED);
+	if ( depr_attr )
+		{
+		ConstExpr* expr = (ConstExpr*)depr_attr->AttrExpr();
```

Comment:
Is there a reason to use the C-style cast rather than `static_cast<ConstExpr*>` here? Seems like it couldn't hurt to lean on the compiler a little, but I could be missing something :)

Reply:
Nope, it can't definitely be a c++-style cast.